pkg/cli/lambda_server__added.go

```diff
@@ -77,12 +77,14 @@ func OptionallyAddLambdaCommand[T field.Configurable](
 		if err != nil {
 			return err
 		}
-		defer func() {
-			err := otelShutdown(context.Background())
-			if err != nil {
-				zap.L().Error("error shutting down otel", zap.Error(err))
-			}
-		}()
+		if otelShutdown != nil {
+			defer func() {
+				err := otelShutdown(context.Background())
```

Comment:
should this itself have a 5 second shutdown or something?

Reply:
updated